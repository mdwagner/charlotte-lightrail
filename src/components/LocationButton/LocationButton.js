import React from 'react';
import PropTypes from 'prop-types';
import { LocationMarkerTouchableOpacity, IconImage } from './LocationButtonCss';

const errorIcon = require('../../assets/icons/location/ic_location_disabled_white.png');
const loadingIcon = require('../../assets/icons/location/ic_location_searching_white.png');
const locationIcon = require('../../assets/icons/location/ic_my_location_white.png');

export default class LocationButton extends React.Component {

  static propTypes = {
    error: PropTypes.string,
    loading: PropTypes.bool.isRequired,
    seeAllStations: PropTypes.func.isRequired
  }

  getLocationImage() {
    const { error, loading } = this.props;
    if (error) {
      return (
        <IconImage
          inactiveIcon
          source={errorIcon}
        />
      );
    } else if (loading) {
      return (
        <IconImage
          inactiveIcon
          source={loadingIcon}
        />
      );
    }
    return (
      <IconImage source={locationIcon} />
    );
  }

  render() {
    const { seeAllStations } = this.props;
    return (
      <LocationMarkerTouchableOpacity onPress={seeAllStations}>
        {this.getLocationImage()}
      </LocationMarkerTouchableOpacity>
    );
  }
}
