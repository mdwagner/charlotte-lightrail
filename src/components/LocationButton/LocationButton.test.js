import React from 'react';
import renderer from 'react-test-renderer';
import LocationButton from './LocationButton';

describe('<LocationButton />', () => {
  let props;
  let loading;

  beforeAll(() => {
    props = {
      seeAllStations: jest.fn()
    };
    loading = true;
  });

  it('renders correctly with an error', () => {
    const error = 'Error message';

    const tree = renderer.create(
      <LocationButton {...props} error={error} loading={loading} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with no error and loading', () => {
    const tree = renderer.create(
      <LocationButton {...props} loading={loading} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with no error and not loading', () => {
    const tree = renderer.create(
      <LocationButton {...props} loading={false} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
