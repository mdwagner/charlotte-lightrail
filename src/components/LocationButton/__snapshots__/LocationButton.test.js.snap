exports[`<LocationButton /> renders correctly with an error 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "flex-end",
      "flexBasis": 0,
      "flexGrow": 0.8,
      "flexShrink": 1,
      "opacity": 1,
      "paddingRight": 14,
      "paddingTop": 1,
    }
  }
  testID={undefined}>
  <Image
    inactiveIcon={true}
    source={1}
    style={
      Array [
        Object {
          "opacity": 0.4,
        },
        undefined,
      ]
    } />
</View>
`;

exports[`<LocationButton /> renders correctly with no error and loading 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "flex-end",
      "flexBasis": 0,
      "flexGrow": 0.8,
      "flexShrink": 1,
      "opacity": 1,
      "paddingRight": 14,
      "paddingTop": 1,
    }
  }
  testID={undefined}>
  <Image
    inactiveIcon={true}
    source={1}
    style={
      Array [
        Object {
          "opacity": 0.4,
        },
        undefined,
      ]
    } />
</View>
`;

exports[`<LocationButton /> renders correctly with no error and not loading 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "flex-end",
      "flexBasis": 0,
      "flexGrow": 0.8,
      "flexShrink": 1,
      "opacity": 1,
      "paddingRight": 14,
      "paddingTop": 1,
    }
  }
  testID={undefined}>
  <Image
    source={1}
    style={
      Array [
        Object {
          "opacity": 1,
        },
        undefined,
      ]
    } />
</View>
`;
